### ISSUE TITLE
docparams property type check is inverted

### ISSUE THREAD

**Issue by StephenBrown2** [PyCQA/pylint#2883]
Hi there!

It appears that the check for a property type is inverted here:
https://github.com/PyCQA/pylint/blob/377e82fdf21c74fe497217bfc97f696b36ead233/pylint/extensions/docparams.py#L299

When I believe it should be:
```python
if not (doc.has_rtype() or (is_property and not doc.has_property_type())):
```
Thus checking if there doesn't exist a `:type:` doc on a `@property` function.

### Steps to reproduce
1. Create `bad.py`:
```python
class MyObjectClass:
    pass

class MyClass:
    @property
    def my_func(self):
        """ Object

        :type: :class:`~bad.MyObjectClass`

        :raises ValueError: The value is not supported.

        Also raises all exceptions from :func:`~other_func`
        """
        return MyObjectClass
```
2. Run pylint
3. Observe results

It should be noted that this does not trigger on plain types, like `str`, but I think that is a separate issue.

### Current behavior
```
bad.py:6: [W9012(missing-return-type-doc), MyClass.my_func] Missing return type documentation
```

### Expected behavior
No output

### pylint --version output

```
pylint 2.3.1
astroid 2.2.5
Python 3.7.3 (default, Mar 26 2019, 21:43:19) 
[GCC 8.2.1 20181127]
```

**Comment by PCManticore** [comment]
Hey @StephenBrown2 Thanks for reporting an issue. In this case the check is correct as it is expecting either a `rtype` or an `returntype` annotation, which your example does not provide: https://github.com/PyCQA/pylint/blob/377e82fdf21c74fe497217bfc97f696b36ead233/pylint/extensions/_check_docs_utils.py#L406

**Comment by StephenBrown2** [comment]
But the function is decorated with `@property`, which makes it behave as a read-only attribute, rather than a class/instance function.

Thus, `is_property` should be `True`, resulting in the following evaluation for the if statement:

```python
if not (doc.has_rtype() or (doc.has_property_type() and is_property)): 
>>
if not (False or (True and True)):
>>
if not (False or True):
>>
if not (True):
>>
if False:
```
My suggested restructure would make the check be:
```python
if not (doc.has_rtype() or (is_property and not doc.has_property_type())):
>>
if not (False or (True and not True)):
>>
if not (False or (True and False)):
>>
if not (False or False):
>>
if not False:
>>
if True:
```
The main benefit being checking for `@property` becoming more important and overriding the regular function def check.
